src/formats/ff.c

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+// farbfeld format decoder
+
+#include "../loader.h"
+#include <netinet/in.h>
+#include <string.h>
+
+enum loader_status decode_ff(struct image* ctx, const uint8_t* data,
+                              size_t size)
+{
+    size_t width, height;
+
+    if (size < 16 ) {
+        return ldr_unsupported;
+    }
+
+    static uint8_t magic[8] = {'f', 'a', 'r', 'b', 'f', 'e', 'l', 'd'};
+
+    if (memcmp(data, magic, 8) != 0) return ldr_unsupported;
+
+    memcpy(&width, (const uint8_t*) data+8, 4);
```

Comment:
It's not necessary, `width = *(const uint32_t*) & data[8]`.
But it would be better i if you define a simple structure something like this:
```
struct farbfeld_header {
    uint8_t magic[8];
    uint32_t width;
    uint32_t height;
};
```